containerfixture/containerfixture.go

```diff
@@ -0,0 +1,118 @@
+package containerfixture
+
+import (
+	_ "embed"
+	"fmt"
+	"log"
+	"math/rand"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+//go:embed containerfixture.yaml
+var manifest string
+
+const Username = "admin"
+const Password = "Password"
+
+// Generates a random dynamic port number
+func GetRandomPort() string {
+	minPort := 49152
+	maxPort := 65535
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+	randomPort := fmt.Sprintf("%d", r.Intn(maxPort-minPort)+minPort)
+	return randomPort
+}
+
+// Modify the manifest to allow running it in parallel as
+// many times as needed with dynamic values
+func modifyYaml(yaml string) map[string]string {
+	podName := fmt.Sprintf("test-pod-%s", GetRandomPort())
+	webPort := GetRandomPort()
+	apiPort := GetRandomPort()
+
+	yaml = strings.Replace(yaml, "name: \"test-pod\"", fmt.Sprintf("name: %s", podName), 1)
+	yaml = strings.Replace(yaml, "hostPort: 8000", fmt.Sprintf("hostPort: %s", webPort), 1)
+	yaml = strings.Replace(yaml, "hostPort: 8089", fmt.Sprintf("hostPort: %s", apiPort), 1)
+
+	return map[string]string{
+		"yaml":    yaml,
+		"podName": podName,
+		"webPort": webPort,
+		"apiPort": apiPort,
+	}
+}
+
+// Building and Running a k8s manifest using 'podman kube play' command.
+// It will make up to three attempts to run with different configurations,
+// allowing it to run concurrently if necessary
+func BuildAndRunPod() map[string]string {
+	var deployment map[string]string
+	attempts := 3
+	for attempt := 0; attempt < attempts; attempt++ {
+		deployment := modifyYaml(manifest)
+
+		cmd := exec.Command("podman", "play", "kube", "-")
+		cmd.Stdin = strings.NewReader(deployment["yaml"])
+		output, err := cmd.CombinedOutput()
+		if err == nil {
+			outputString := strings.Split(strings.TrimSpace(string(output)), "\n")
+			containerID := outputString[len(outputString)-1]
+			deployment["containerID"] = containerID
+			return deployment
+		}
+		log.Printf("Failed to start pod (attempt %d of %d).\n", attempt+1, attempts)
+	}
+	Cleanup("", deployment["podName"])
+	log.Fatalf("Failed to start pod after %d attempts", attempts)
+	return make(map[string]string) // Avoid 'missing return' error
+}
+
+// VerifySplunkServiceIsUp continuously monitors the Splunk container to
+// ensure it is up and ready for use in tests. It achieves this by repeatedly
+// checking the status endpoint of the Splunk API over a period of two minutes.
+func VerifySplunkServiceIsUp(port string) {
```

Comment:
This could have a generalized name sine there is nothing Splunk-specific here.

Reply:
The endpoint itself is Splunk-specifc, as well as most of the cmd used (`fmt.Sprintf("%s:%s", Username, Password), "--insecure", splunkURL)`). modifying this alongside the logging make the function too messy iMO

Reply:
that Sprintf shouldn't be needed - we should have a netrc file taking care of that - easier and more reliable to have an env var pointing to a static file , and we would need the same for the test anyway.

The endpoint URL could be passed as a parameter.